### PR TITLE
tikv: document backup.num-threads config (#4057)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #4057 to release-3.1

---

### What is changed, added or deleted? (Required)

Document the new TiKV configuration `backup.num-threads` introduced in tikv/tikv#8193.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by TomShawn** [comment]
@kennytm Please resolve the conflict in this PR.

**Comment by kennytm** [comment]
PTAL @TomShawn

**Comment by kennytm** [comment]
```
npm ERR! code ENOTFOUND
npm ERR! errno ENOTFOUND
npm ERR! network request to https://registry.npmjs.org/markdownlint-cli failed, reason: getaddrinfo ENOTFOUND registry.npmjs.org
```

spurious network error

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -1082,6 +1082,16 @@ raftdb 相关配置项。
 + 默认值：8
 + 最小值：1
 
+## backup
+
+用于 BR 备份相关的配置项。
+
+### `num-threads`
+
++ 处理备份的工作线程数。
```
Comment (TomShawn) [inline tikv-configuration-file.md:1091]:
```suggestion
+ 处理备份的工作线程数
```

**Pull request merged**